File: POProjekt3/rosliny/Trawa.py
from Roslina import Roslina

class Trawa(Roslina):
    
    def __init__(self):
        pass

    def __init__(self, swiat, poz, wiek, inicj, moc):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeWiek(wiek)
        super().changeInicjatywa(inicj)

        s = "T"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    def __init__(self, swiat, poz):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeInicjatywa(0)
        super().zmienSile(0)
        super().changeWiek(0)

        s = "T"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    #--------------------------------------------

    def rysowanie(self):
        print('T')

    def getZnak(self):
        return 'T'

File: VirtualWorld-Cpp/Virtual-World/Trawa.cpp
#include "Owca.h"
#include "Zwierze.h"
#include <iostream>
#include "Organizm.h"
#include "Trawa.h"
#include "Roslina.h"

Trawa::Trawa()
{
}

Trawa::Trawa(Swiat* swiat, std::vector<unsigned int> poz, int wiek, int inicj, int moc)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeWiek(wiek);
	changeInicjatywa(inicj);

	getSwiat()->edytujPlansze(poz[0], poz[1], TRAWA);
}

Trawa::Trawa(Swiat* swiat, std::vector<unsigned int> poz)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeInicjatywa(0);
	zmienSile(0);
	changeWiek(0);

	getSwiat()->edytujPlansze(poz[0], poz[1], TRAWA);
}

void Trawa::rysowanie()
{
	std::cout << TRAWA;
}

char Trawa::getZnak()
{
	return TRAWA;
}

Trawa::~Trawa()
{
}